src/datasets/exceptions.py

```diff
@@ -5,10 +5,23 @@
 class DatasetsError(Exception):
     """Base class for exceptions in this library."""
 
-    pass
-
 
 class DefunctDatasetError(DatasetsError):
     """The dataset has been defunct."""
 
-    pass
+
+class FileNotFoundDatasetsError(DatasetsError, FileNotFoundError):
+    """FileNotFoundError raised by this library."""
+
+
+class DataFilesNotFoundError(FileNotFoundDatasetsError):
+    """No (supported) data files found."""
+
+
+class DatasetNotFoundError(FileNotFoundDatasetsError):
+    """Dataset not found.
+
+    Raised when trying to access:
+    - a missing dataset, or
+    - a private/gated dataset and the user is not authenticated.
+    """
```

Comment:
Maybe we could re-use `huggingface_hub`'s `RepositoryNotFound` and `GatedRepo` errors instead of introducing our own (this exception should at least subclass them)

For example, `transformers` throws an `EnvironmentError` (with the error description)  and chains the caught (`huggingface_hub`) exception, so consistency with them would be nice.

Reply:
I agree with subclassing `huggingface_hub` errors as well, but at the same time our `datasets` `DatasetsError` base clase, so that a user of `datasets` can catch all `datasets` errors by using this `DatasetsError` class.

Reply:
I would propose to address in a subsequent PR to catch specific `huggingface_hub` errors (instead of parsing the HTTP error status code: this is done by `huggingface_hub`) and raise our specific errors accordingly.